### PR TITLE
Improve offline mode react performance

### PR DESCRIPTION
**Pull request by sliptype**
### Description

Improves selectors used in `DownloadStatusIndicator` and `Audio` to minimize rerenders while offline mode is on.
0 rerenders on feed screen while offline mode is on!

Will follow up with a pr to address excess image fetching and rn queue performance

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/components/audio/Audio.tsx [hunk 1]
```diff
@@ -117,24 +118,40 @@ export const Audio = () => {
...
 
   // Queue Things
   const queueIndex = useSelector(getIndex)
   const queueShuffle = useSelector(getShuffle)
   const queueOrder = useSelector(getOrder)
   const queueTrackUids = queueOrder.map((trackData) => trackData.uid)
-  const queueTrackMap = useSelector((state) =>
-    getTracks(state, { uids: queueTrackUids })
+  const queueTrackIds = queueOrder.map((trackData) => trackData.id)
+  const queueTrackMap = useSelector(
```
Comment (sliptype) [inline packages/mobile/src/components/audio/Audio.tsx:128]: `useProxySelector` was still rerendering, not sure why

**Review by amendelsohn — APPROVED** [review]
This is great!

Wondering if we should audit usage of the other selectors. With all these keyed objects, I feel like there are other places I'm selecting the whole object where I shouldn't. Better to make these where we pass in the ids, right?

**Comment by sliptype** [comment]
> This is great!
> 
> Wondering if we should audit usage of the other selectors. With all these keyed objects, I feel like there are other places I'm selecting the whole object where I shouldn't. Better to make these where we pass in the ids, right?

Yes! I'm going through them now

File: packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx [hunk 2]
```diff
@@ -27,10 +27,10 @@ export const AlbumsTab = () => {
   const [filterValue, setFilterValue] = useState('')
   const isReachable = useSelector(getIsReachable)
   const isOfflineModeEnabled = useIsOfflineModeEnabled()
-  const offlineDownloadStatus = useSelector(getOfflineDownloadStatus)
   const userAlbums = useProxySelector(
-    (state: CommonState) =>
-      getAccountCollections(state, filterValue).filter((collection) => {
+    (state: AppState) => {
+      const offlineDownloadStatus = getOfflineDownloadStatus(state)
```
Comment (amendelsohn) [inline packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx:32]: Ah, nice. So this rolls it all into one dependency?

How does this work if the status changes? I assume the hook runs but the component only rerenders if the result is different?

**Review by amendelsohn — APPROVED** [review]

File: packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx (hunk 2 above)
Comment (sliptype) [reply packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx:32]: Exactly! Any time the store state changes _all_ selectors in useSelectors are rerun and the result compared by default with reference equality

useProxySelector does this slightly more efficiently by tracking which parts of the state got accessed and then only rerendering if those parts changed

Comment (amendelsohn) [reply packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx:32]: Perfect. That's super cool. So we should be combining our complex selectors whenever possible.

Comment (sliptype) [reply packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx:32]: It's definitely a tradeoff, running a more complex computation every time the state changes vs rerendering a component more often. Usually rendering react components is wayyy more expensive though. Also there is some overhead to useProxySelector but I'm not sure how big

**Pull request merged**